File: 14-LongestCommonPrefix/14-LongestCommonPrefix.java
// Last updated: 4/24/2026, 8:15:26 AM
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int minLenStr = 201;
        for (String str : strs) {
            minLenStr = Math.min(minLenStr, str.length());
        }
        String ans = "";
        for (int i = 0; i < minLenStr; ++i) {
            for (String str : strs) {
                if (str.substring(0, i + 1).equals(strs[0].substring(0, i + 1)))
                    continue;
                else
                    return ans; 
            }
            ans = strs[0].substring(0, i + 1);
        }
        return ans;
    }
}

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.cpp
// Last updated: 4/24/2026, 8:15:29 AM
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string lcp = "";
        string firstStr = strs[0];
        for (int i = 0; i < firstStr.size(); ++i) {
            bool flag = true;
            for (int j = 1; j < strs.size(); ++j) {
                if (i + 1 > strs[j].size() || firstStr[i] != strs[j][i]) {
                    goto outer;
                }
            }
            if (flag) lcp += firstStr[i];
        }
        outer:
        return lcp;
    }
};